java-bridge-forsyde-io/src/main/java/idesyde/forsydeio/MM_MCoreAndPL_IRule.java

```diff
@@ -0,0 +1,254 @@
+						me.getViewedVertex())));
+		// basically this check to see if there are always neighboring
+		// pe, mem and ce
+		// and also the subset of only communication elements
+		var processorsProvisions = new HashMap<String, Map<String, Map<String, Double>>>();
+		for (var pe : processingElements) {
+			ForSyDeHierarchy.InstrumentedProcessingModule
+					.tryView(pe)
+					.ifPresent(ipe -> processorsProvisions.put(pe.getIdentifier(),
+							ipe
+									.modalInstructionsPerCycle()));
+		}
+		// var processorsProvisions = processingElements.map(pe -> {
+		// // we do it mutable for simplicity...
+		// // the performance hit should not be a concern now, for super big instances,
+		// this can be reviewed
+		// var mutMap = new HashMap<String, Map<String, Double>>();
+		// return mutMap;
+		// });
+		if (processingElements.size() <= 0) {
+			errors.add("MM_McoreAndPL_IRule: no processing elements");
+		}
+		if (memoryElements.size() <= 0) {
+			errors.add("MM_McoreAndPL_IRule: no memory elements");
+		}
+		if (plElements.size() <= 0) {
+			errors.add("MM_McoreAndPL_IRule: no logic programmable elements");
+		}
+		if (!processingOnlyValidLinks || !memoryOnlyValidLinks) {
+			errors.add("MM_McoreAndPL_IRule: processing or memory have invalid links");
+		}
+		if (!pesConnected) {
+			errors.add("MM_McoreAndPL_IRule: not all processing elements reach a memory element");
+		}
+		if (processingElements.size() > 0 &&
+				memoryElements.size() > 0 &&
+				processingOnlyValidLinks &&
+				memoryOnlyValidLinks &&
+				pesConnected) {
+			var interconnectTopologySrcs = new ArrayList<String>();
+			var interconnectTopologyDsts = new ArrayList<String>();
+			topology
+					.edgeSet()
+					.forEach(e -> {
+						interconnectTopologySrcs.add(topology.getEdgeSource(e)
+								.getIdentifier());
+						interconnectTopologyDsts.add(topology.getEdgeTarget(e)
+								.getIdentifier());
+					});
+			var procFreqs = processingElements.stream().collect(Collectors.toMap(
+					pe -> pe.getIdentifier(),
+					pe -> pe.operatingFrequencyInHertz()));
+			var memSizes = memoryElements.stream()
+					.collect(Collectors.toMap(
+							me -> me.getIdentifier(),
+							me -> me.spaceInBits()));
+			var maxConcurrentFlits = communicationElements.stream()
+					.collect(Collectors.toMap(
+							ce -> ce.getIdentifier(),
+							ce -> ForSyDeHierarchy.InstrumentedCommunicationModule
+									.tryView(ce)
+									.map(ice -> ice.maxConcurrentFlits())
+									.orElse(1)));
+			var bandwidths = communicationElements.stream()
+					.collect(Collectors.toMap(
+							ce -> ce.getIdentifier(),
+							ce -> ForSyDeHierarchy.InstrumentedCommunicationModule
+									.tryView(ce)
+									.stream()
+									.mapToDouble(ice -> (double) ice
+											.flitSizeInBits()
+											* (double) ice.operatingFrequencyInHertz()
+											/ (double) ice.maxCyclesPerFlit())
+									.max()
+									.orElse(0.0)));
+			var totalAvailablePLAreas = plElements.stream()

```

Comment:
@Rojods extract the area from each of the components that should be fed to the decision model